pages/data.js

```diff
@@ -131,14 +134,18 @@ export default class Explore extends React.Component {
       const uniqueRecords = [...new Set(newData.records[key])];
       uniqueRecords.forEach(record => (filters[key][record] = true));
     });
-    this.setState({
+    this.setState(prevState => ({
+      isLoading: false,
       activeDataset: selectedDataset,
       data: {
-        ...data,
-        [selectedDataset]: newData,
+        ...prevState.data,
```

Comment:
This fixes a violation of [State Updates May Be Asynchronous](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous).